test/tests/functional/pbs_pbsnodes_attributes.py

```diff
@@ -0,0 +1,115 @@
+# trademark licensing policies.
+
+from tests.functional import *
+
+
+class Testpbsnodes_attributes(TestFunctional):
+
+    def Test_pbsnodes(self, user):
+        """
+        Function to validate pbsnodes attribute on user basis
+        """
+
+        attr_dict = {}
+        host_short_name = self.mom.shortname
+        host_name = self.mom.hostname
+        pbs_version = self.server.pbs_version
+
+        command = os.path.join(self.server.pbs_conf['PBS_EXEC'],
+                               'bin', 'pbsnodes -a')
+        if user == 'root':
+            ret = self.du.run_cmd(self.server.hostname, command, sudo=True)
+        else:
+            ret = self.du.run_cmd(self.server.hostname, command, runas=user)
+
+        #converting pbsnodes -a output into dictionary for validation
+        attr_list = ret['out']
+        list_len = len(attr_list) - 1
+        attr_list = filter(None, attr_list)
+        for i in range(1, list_len):
+            attr = attr_list[i].split('=')[0].strip()
+            val = attr_list[i].split('=')[1].strip()
+            attr_dict[attr] = val
+
+        if 'Mom' in attr_dict:
+            self.assertEqual(attr_dict['Mom'], host_name)
+        if 'ntype' in attr_dict:
+            self.assertEqual(attr_dict['ntype'], 'PBS')
+        if 'state' in attr_dict:
+            self.assertEqual(attr_dict[ATTR_NODE_state], 'free')
+        if 'resources_available.arch' in attr_dict:
+            self.assertEqual(attr_dict['resources_available.arch'], 'linux')
+        if 'resources_available.vnode' in attr_dict:
+            self.assertEqual(attr_dict['resources_available.vnode'],
+                             host_short_name)
+        if 'resources_available.host' in attr_dict:
+            self.assertEqual(attr_dict['resources_available.host'],
+                             host_short_name)
+        if 'resources_assigned.accelerator_memory' in attr_dict:
+            self.assertEqual(attr_dict
+                             ['resources_assigned.accelerator_memory'], '0kb')
+        if 'resources_assigned.hbmem' in attr_dict:
+            self.assertEqual(attr_dict['resources_assigned.hbmem'], '0kb')
+        if 'resources_assigned.mem' in attr_dict:
+            self.assertEqual(attr_dict['resources_assigned.mem'], '0kb')
+        if 'resources_assigned.naccelerators' in attr_dict:
+            self.assertEqual(attr_dict['resources_assigned.naccelerators'],
+                             '0')
+        if 'resources_assigned.ncpus' in attr_dict:
+            self.assertEqual(attr_dict['resources_assigned.ncpus'], '0')
+        if 'resources_assigned.vmem' in attr_dict:
+            self.assertEqual(attr_dict['resources_assigned.vmem'], '0kb')
+        if 'resv_enable' in attr_dict:
+            self.assertEqual(attr_dict['resv_enable'], 'True')
+        if 'sharing' in attr_dict:
+            self.assertEqual(attr_dict['sharing'], 'default_shared')
```

Comment:
Also please run this on cray simulator or real CRAY. I think we are missing few more attributes here.
Same thing for UV and ICE.